doc/user/content/sql/drop-type.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "DROP TYPE"
+description: "`DROP TYPE` removes a user-defined data type."
+menu:
+  main:
+    parent: 'sql'
+---
+
+{{< version-added v0.7.1 />}}
+
+`DROP TYPE` removes a [custom data type](../create-type). You cannot use it on default data types.
+
+## Syntax
+
+{{< diagram "drop-type.svg" >}}
+
+Field | Use
+------|-----
+_data_type_name_ | The name of the type to remove.
+`CASCADE` | Automatically removes objects that depend on the type, such as tables or other types.
+`IF EXISTS`  |  Issue a notice, but not an error, if the type doesn't exist.
+`RESTRICT`  |  Refuse to remove the type if any objects depend on it. This is the default.
+
+
+## Examples
+
+### Remove a data type with no dependencies
+```sql
+CREATE TYPE int4_map AS MAP (key_type=text, value_type=int4);
+
+SHOW TYPES;
+```
+```
+CREATE TYPE
+    name
+--------------
+  int4_map
+(1 row)
+```
+
+```sql
+DROP TYPE int4_map;
+SHOW TYPES;
+```
+```
+DROP TYPE
+  name
+--------------
+(0 rows)
+```
+
+### Remove a data type with dependent objects
+
+{{< warning >}}
+By default, `DROP TYPE` will not remove a type with dependent objects. The `CASCADE` switch will remove both the specified type and *all its dependent objects*.
+{{< /warning >}}
+
+In the example below, the `CASCADE` switch removes `int4_list`, `int4_list_list` (which depends on `int4_list`), and the table *t* which has a column of data type `int4_list`.
+
+```sql
+CREATE TYPE int4_list AS LIST (element_type = int4);
+
+CREATE TYPE int4_list_list AS LIST (element_type = int4_list);
+
+CREATE TABLE t (a int4_list);
+
+SHOW TYPES;
+```
+```
+CREATE TYPE
+CREATE TYPE
+CREATE TABLE
+      name
+----------------
+ int4_list
+ int4_list_list
+(2 rows)
+```
+
+```sql
+DROP TYPE int4_list CASCADE;
```

Comment:
Oh, are you trying to show some series of commands and then their output? I found this really non-intuitive and would really suggest following the examples from the rest of the docs and showing the input and the output sequentially.

Reply:
But that is how we're already doing it on multiple other pages. See CREATE TYPE and SHOW SOURCES for examples.

Reply:
I am looking at [`CREATE TYPE`](https://materialize.com/docs/sql/create-type/) and that is not how the code samples are formatted.

Those code samples are formatted as:

```
CREATE TYPE...

SELECT some value of that type...
```

They do not include the response of the statement that just executed as you have below with e.g. `DROP TYPE`. There is also no interleaving of responses.

The code samples of `SHOW SOURCES` are also not interleaved.

So––not sure what you're seeing but it differs from what I'm seeing both live on the site and in my repo.

Reply:
All of the `CREATE TYPE` examples show:

```
CREATE TYPE...
SELECT TYPE...
Results of the select type
```
without any separation of the `SELECT` command and the responses. They do skip the `CREATE TYPE` success commands, which doesn't make sense to me; either the entire relevant response should be included, or none of it should.

Reply:
That is certainly a valid opinion, but I don't think that it's accurate to say that anything in the existing docs concatenates all of the inputs followed by all of the outputs. I also don't think that is the format we should use for our code samples.

Reply:
We don't currently seem to have a standard format for this. In four different pages, there are four different formats:

**ALTER RENAME:**

First field: 
```
Command
Output
```

Second field:
```
Command
Command (show results of previous command)
Output
```

**CREATE DATABASE:**

First field:
`Command`

Second field:
```
Command
Output
```

**CREATE TABLE:**

First field:
`Command`

Second field:
```
Command (show results of command from previous field)
output

Command (show different results of command from previous field)
output
```

**CREATE TYPE:**
In a single field:

```
Command

Command
Output
```

We're also not particularly consistent about whether we show the output when it's just a command confirmation (eg `DROP SOURCE`), although in that case there's a strong trend towards not including it.

I'll put together a style guide for this, but any decision here is going to be inconsistent with some of our docs until I have time to standardize them.

Reply:
Consistent formats for code examples, as long as any given page is understandable/correct, is **superbly** unimportant. I would so much rather see actual documentation being produced than coming up with a style guide that requires someone to enforce it. Like, I cannot stress how much more important producing documentation is than coming up with stringent guidelines. The comment here was just that the format you chosen was unintuitive; no need to take on making all of them consistent until we start hearing from users that there is a problem.

Reply:
And I strongly disagree. The format should be consistent across a site so that users do not have to waste any time or mental effort to figure out how things are organized, or what is included or excluded, once they know the general pattern of the documentation.

Reply:
Our code samples are so trivial that anyone familiar with SQL will have no problem understanding the ever-so-slightly different formats. However, if you don't document features, no one can use them irrespective of how consistent your other code samples are.

You have to choose the things you work on because of the finiteness of time. Privileging a task like making trivial code samples consistent––without any signal from users that there is _any problem whatsoever_ ––is counterproductive. If you can substantiate that there is a problem, or if you don't have new docs to create, then knock yourself out. But, I can tell you, as someone who reads ton of code samples that the varied formats across all documentation sites nominally, if at all, impacts my ability to understand what the code is driving at. However, when I try to work with an undocumented API, it is nearly impossible.

Reply:
How is using a consistent code format going forward any different from making sure that we use "drop" or "remove" consistently when describing an action?

Reply:
Two big things:

- The amount of effort required. The point I am making to you about the unimportance of code formatting is to stress the fact that spending time to develop a style guide for code samples is not a wise use of time given the opportunity cost of creating other, more impactful documentation. In the fullness of time, consistent code samples would be great, but it is polish that you should be concerned with only once you have the structure in working order. However, simply changing a few words in a PR that is already under review presents almost no opportunity cost. That alone is a dramatic difference.
- The words in question ("drop" and "remove") _could_ have semantic differences, which introduce ambiguities that the user could have a difficult time parsing out. This same line of reasoning simply doesn't apply to a code sample––the beautiful thing about code samples is that they're unambiguous if they're correct. You could make an argument that not including the tautological response for some commands is confusing to some users, but you would have a hard time making the case that their absence introduces an ambiguity for someone following along. So, I would say that these two actions also fall under different classes of issue.